### ISSUE TITLE
Missing $CONDA_PREFIX/MetaLAFFA/config/steps?

### ISSUE THREAD

**Issue by padbc** [borenstein-lab/MetaLAFFA#7]
Dear all,

I am trying to modify the cluster_params parameter, but I can't seem to find the map_read_to_genes.py script -- in fact, the entire config folder is missing from the conda installation. I installed the MetaLAFFA using mamba, btw. I would appreciate any input you may have.

**Comment by engal** [comment]
Sorry, if you used the -u flag, did you get any additional output beyond

```
Warning: The gene-to-ortholog table (/home/pedro/miniconda3/envs/metalaffa/MetaLAFFA/gene_to_ortholog_maps/uniref90_to_ko.map) does not exist. You will be unable to perform the default ortholog mapping step (and the hit filtering step if using either the 'best_ortholog' or 'best_n_orthologs' methods) without it. Make sure that 'gene_to_ortholog_directory' in config.file_organization and 'gene_to_ortholog_file' in config.operation.py are correct.
End of database preparation.
```
There should be more including messages saying that it is performing certain steps and/or skipping unnecessary steps. Did you get any other messages, or just that warning?

**Comment by engal** [comment]
You're correct, it looks like they removed KO mappings from the current version of the file on UniProt. They still have mappings for organism-specific genes in the KEGG database, but not ortholog classifications. This means that the latter steps of MetaLAFFA (ortholog abundance correction and ortholog aggregation) will no longer work under the default configuration since the default approaches rely on KO IDs.

Regarding other ortholog types, you can map to any database that exists in the second column of the idmapping.dat.gz file. For example, you can map to RefSEq, eggNOG, BioCyc, etc. To create the mapping table, first go the the operation.py submodule in the config module and change `target_ortholog` to your new choice (line 95). Then rerun the database preparation script, and that should generate the new mapping file.

**Comment by padbc** [comment]
Thank you, @engal. Since the main MetaLAFFA approach is predicated upon the existence of KO annotations, would it make sense to make this clear on the software's landing page, and potentially suggest an alternative?

With regard to the UniProt-KO cross-references, I will contact UniProt to see what's happening. While using an outdated uniprot-ko mapping is not ideal, I wonder how big of an impact this would have, considering that KOs are aggregated into pathways/modules.

**Comment by engal** [comment]
Quick update, I heard back from UniProt and it is indeed a KEGG licensing issue that led them to remove their KO mappings. I'm working on an update to MetaLAFFA that will switch to using the last available version of UniProt where KO mappings were available. I'll update this issue once the fix is done.

**Comment by engal** [comment]
Sorry for the delay, but I've now modified the default prepare_databases.py script to instead use the most recent version of the UniProt database that also includes KO mappings (from 2020). Now KO mappings should always be available when using the default installation, assuming UniProt does not modify their archives.

These changes are in the most recent MetaLAFFA release (1.0.1). Please let me know if you have any further issues.

**Comment by padbc** [comment]
Thank you for letting me know. I will test this shortly. How recently has this new version been tested? The https://ftp.uniprot.org/pub/databases/uniprot/previous_releases/ site now appears to be empty.

**Comment by engal** [comment]
I'm not sure why you are unable to see the contents, though when I first clicked the address you posted, nothing showed up, but after refreshing the page was fine. This component of the package was tested last week, so unless they made any changes since then, it should be fine. Wget still works to access the contents that should be there.
